### ISSUE TITLE
Every group reply creates a new conversation

### ISSUE THREAD

**Issue by jmdodge** [klinker-apps/messenger-issues#406]
I'm running a Samsung Galaxy S8+ with the latest version of the app. Android version 7.0

1. Create a new conversation - recipients John, Joe.
2. Send a message. So far so good, I have one conversation.
3. When a reply is received, it does not go into the conversation I started. A new conversation has been created with participants John, Joe, and _me_. This second conversation is where I see the reply.
4. If I type a reply to the reply (in the 2nd conversation that was created when I received the reply), I will receive an additional copy of the message I sent. This additional copy creates _yet another conversation_ - this one has John, Joe, me, and me ("me" is listed twice).

I now have *three* conversations when I just tried to start one.

**Comment by klinker24** [comment]
Is it your name that is listed or just your phone number, or what?

**Comment by jmdodge** [comment]
It is literally the word "Me".
When I view the participant details of the conversation, I see the names and phone numbers of the other participants along with the word "me" and my own phone number.

**Comment by klinker24** [comment]
So it seems like Pulse doesn't know what your phone number is and isn't stripping it out of the messages. 

If you open the navigation drawer, does it display your phone number correctly in the phone app? Do you use the online service? If so, you can change your phone number manually to reflect what it actually is.

**Comment by klinker24** [comment]
Changes with: https://github.com/klinker41/messenger/commit/e28f5f845e2d9fcd62d6e673ff3c5a18c02154fc

I think that these changes should take care of your issue. They will be released with the next update for Pulse. You might have to delete that conversation and start it fresh to see the changes take hold. 

If you don't want to delete the conversation:

* if you use the online Pulse service: use "My account" -> "Resync Messages"
* if you don't use the online service: uninstall and reinstall the app

**Comment by jmdodge** [comment]
A little follow up here: based on the the information you provided, I think I was able to resolve on my own:
1. I checked the hamburger menu in Pulse and confirmed that I did see my own phone number.
2. I went into the Android contacts app and edited the profile for my own number on the main page of the fly out menu. I noticed there were two phone numbers - the first one was a temporary number that had been assigned to my phone while my real number was porting. It was listed first. I removed it.
3. I went through my contacts and I had two google contacts that had my own number - one with my name "Justin" and one titled "Me". I deleted them both.

These steps seem to have resolved the issue. I'm guessing the real problem was in step 2.
